### DISCUSSION TITLE
SHAP for logistic regression: Masker on train or test set?

### DISCUSSION THREAD

**Discussion by ghost** [shap/shap#1867] [Q&A]
Hi,

I want to know the important features of the test set, but I am not sure whether I need to fit the masker on the training or the test set? Also, what is exactly the difference between de independent and the partition masker?

```
model = LogisticRegression(random_state = 1)
model.fit(X_train, y_train)

masker = shap.maskers.Independent(data = X_train)
**_or_**
masker = shap.maskers.Independent(data = X_test)
```
```

explainer = shap.LinearExplainer(model, masker = masker)
shap_val = explainer(X_test)
```
I would really appreciate it if someone could guide me in this.
